File: jobs/consts.py
from django.db import models


class JobType(models.TextChoices):
    FULL_TIME = "full-time", "Full-time"
    PART_TIME = "part-time", "Part-time"
    CONTRACT = "contract", "Contract"
    INTERNSHIP = "internship", "Internship"


class RemoteOption(models.TextChoices):
    REMOTE = "remote", "Remote"
    ON_SITE = "on-site", "On-site"
    HYBRID = "hybrid", "Hybrid"


# Identifying User-Agent for scrapers that do not need to impersonate a browser.
# Kalibrr, Kitalulus and Karirhub all serve full content to this UA (verified
# 2026-09-09), so those scrapers announce themselves rather than spoofing a TLS
# fingerprint the way the Indeed/LinkedIn scrapers do. See _docs/scraping-policy.md.
BOT_USER_AGENT = "CariinKerjaBot/1.0 (+https://cariinkerja.id)"

File: frontend/src/lib/consts.ts
export const JOB_TYPES = [
  { value: "full-time", label: "Penuh waktu" },
  { value: "part-time", label: "Paruh waktu" },
  { value: "contract", label: "Kontrak" },
  { value: "internship", label: "Magang" },
] as const;

export type JobType = (typeof JOB_TYPES)[number]["value"];

export const REMOTE_OPTIONS = [
  { value: "remote", label: "Remote" },
  { value: "on-site", label: "On-site" },
  { value: "hybrid", label: "Hybrid" },
] as const;

export type RemoteOption = (typeof REMOTE_OPTIONS)[number]["value"];
